.github/workflows/canary-integration-test.yml

```diff
@@ -92,6 +97,14 @@ jobs:
         # copy the test file
         # execute the test file
 
+    - name: Upload canary test result
+      uses: actions/upload-artifact@v2
+      if: failure()
```

Comment:
What if we always uploaded the artifacts instead of conditionally on failure?

Reply:
Done.